www/README.md

```diff
@@ -0,0 +1,121 @@
+### Install dependencies
+
+We have two kinds of dependencies in this project: tools and angular framework code.  The tools help
+us manage and test the application.
+
+* We get the tools we depend upon via `npm`, the [node package manager](https://www.npmjs.com/).
+* We get the angular code via `bower`, a [client-side code package manager](http://bower.io/).
+
+`npm` is configured to automatically run `bower install` and `gulp`. Before you run the application for the first time, simply run this command from the `www/master` directory:
+
+```
+npm install
+```
+
+To start the application, run this command from the `www/master` directory:
+
+```
+npm start
+```
+
+The `gulp` command will start a file watcher which will update the generated `app` code after any changes are saved. Note: gulp file watcher does not currently support adding or deleting files, this will require a restart of gulp). Two new directories will also be created in the project.
+
+* `master/node_modules` - contains npm dependencies
+* `master/bower_components` - contains the angular framework files and any custom dependencies
+
+Bower components should be refernced in one of the `vendor.json` files below:
+
+* `master/vendor.base.json` - 3rd party vendor javascript required to start the app. JS is compiled to `base.js` and loaded before `app.js`
+* `master/vendor.json` - 3rd party vendor scripts to make the app work, usually lazy loaded. Can be js or css. Copied to `vendor/*`.
+
+### Serving the app during development
+
+The app can be served through `kubectl`, but for some types of review a local web server is convenient. One can be installed as follows:
+
+```
+sudo npm install -g http-server
+```
+
+The server can then be launched:
+
+```
+cd app
+http-server -a localhost -p 8000
+```
+
+### Configuration
+#### Configuration settings
+A json file can be used by `gulp` to automatically create angular constants. This is useful for setting per environment variables such as api endpoints.
+*  ```www/master/shared/config/development.json``` or ```www/master/shared/config/production.json``` can be created from the ```www/master/shared/config/development.example.json``` file.
+* ```development.example.json``` should be kept up to date with default values, since ```development.json``` is not under source control.
```

Comment:
Why not? We should keep it in source control IMO, I shouldn't have to touch anything to get a successful UI build.